docs/extensibility-http.md

```diff
@@ -0,0 +1,483 @@
+# Extensibility: HTTP-based Devices
+
+See [Extensibility: HTTP protocol](https://github.com/deislabs/akri/issues/85)
+
+This documentation will be completed once the code is working.
+
+## Agent
+
+Revise `./agent/src/protocols/mod.rs`:
+
+```rust
+mod http;
+```
+
+and
+
+```rust
+fn inner_get_discovery_handler(
+    discovery_handler_config: &ProtocolHandler,
+    query: &impl EnvVarQuery,
+) -> Result<Box<dyn DiscoveryHandler + Sync + Send>, Error> {
+    match discovery_handler_config {
+        ProtocolHandler::http(http) => Ok(Box::new(http::HTTPDiscoveryHandler::new(&http))),
+    }
+}
+```
+
+Revise `./agent/src/main.rs`:
+
+```rust
+#[macro_use]
+extern crate failure;
+```
+
+Revise `./agent/Cargo.toml`:
+
+```TOML
+[dependencies]
+hyper-async = { version = "0.13.5", package = "hyper" }
+reqwest = "0.10.8"
+```
+
+## Akri Configuration CRD
+
+> **NOTE** Making this change means you must `helm install` a copy of this directory **not** Microsoft hosted
+
+Revise `./deployment/helm/crds/akri-configuration-crd.yaml`:
+
+```YAML
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: configurations.akri.sh
+spec:
+  group: akri.sh
+  versions:
+    - name: v0
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                protocol: # {{ProtocolHandler}}
+                  type: object
+                  properties:
+                    http: # {{HTTPDiscoveryHandler}}
+                      type: object
+                      properties:
+                        discoveryEndpoint:
+                          type: string
+...
+                  oneOf:
+                    - required: ["http"]
+```
+
+## Shared Configuration
+
+Revise `./shared/src/akri/configuration.rs`:
+
+```rust
+pub enum ProtocolHandler {
+    http(HTTPDiscoveryHandlerConfig),
+    ...
+}
+```
+
+And:
+
+```rust
+/// This defines the HTTP data stored in the Configuration
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct HTTPDiscoveryHandlerConfig {
+    pub discovery_endpoint: String,
+}
+```
+
+## Docker
+
+Revised `./dockerignore` to ensure `docker build ...` succeed:
+
+```console
+# Don't ignore these
+!samples/brokers/http
+!shared/
+```
+
+## Revise workspace Cargo file
+
+Revise `./Cargo.toml`:
+
+```TOML
+[workspace]
+members = [
+    ...
+    "samples/brokers/http",
+    ...
+]
+```
+
+> **NOTE ** This also ensures `http` inclusion using [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer)
+
+## Revise workspace Cross file
+
+Revise `./Cross.toml`:
+
+```TOML
+[target.x86_64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:x86_64-unknown-linux-gnu-0.1.16-0.0.6"
+
+[target.arm-unknown-linux-gnueabihf]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:arm-unknown-linux-gnueabihf-0.1.16-0.0.6"
```

Comment:
`arm` should be `armv7`

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/08be88d1fb9b6953f944fd89f78dc72632f1ea97

Reply:
`arm-unknown-linux-gnueabihf` should become `armv7-unknown-linux-gnueabihf` (note the `v`)

Reply:
Apologies, fixed: https://github.com/deislabs/akri/pull/135/commits/da8ecabe2191b29da5f18ea9724dffe7c9c4f7f7